pest/src/unicode/mod.rs

```diff
@@ -0,0 +1,284 @@
+//! Character inclusion in binary or General_Category value Unicode sets.
+//!
+//! We rely on dead code elimination to remove the tables that aren't needed.
+
+#![allow(bad_style)]
+
+#[allow(unused)]
+mod binary;
+pub fn ASCII_Hex_Digit(c: char) -> bool {
```

Comment:
This module might be cleaner to read and update as a macro? Something like:

```rust
char_properties! {
    self::binary::ASCII_HEX_DIGIT => ASCII_Hex_Digit,
    self::binary::ALPHABETIC => Alphabetic,
    …
    self::category::UPPERCASE_LETTER => Uppercase_Letter,
}
```

Reply:
That makes a lot of sense. I originally did it by copy/pasting the generated BY_NAME constants and doing regex replace in-editor, but doing the transformation via macro makes for a better presentation. 👍

I'd like to make the macro also spit out `UNICODE_PROPERTY_NAMES`, but that constant needs to be available to meta and derive, whereas the functions are required by the runtime.

Reply:
I plan to write up a script to update the Unicode tries once I figure out the best way to use `ucd-generate` as a library that doesn't add to much maintenance burden to BurntSushi. (Likely it will be caving and just saying "install it".) Hopefully I can also spit out the mod.rs as well.